src/Connections/GithubRepo.ts

```diff
@@ -1258,6 +1261,48 @@ export class GitHubRepoConnection extends CommandConnection<GitHubRepoConnection
             },
         }
     }
+    
+    public static async searchInstallationForRepos(octokit: Octokit, orgName: string, installationId: number, searchTerms?: string) {
+        // First, do a search on GitHub for repos. This will use the user's context so it will find user repos.
+        let searchRepos: string[]|null = null;
+        if (searchTerms) {
+            const terms = encodeURIComponent(searchTerms);
+            const searchResultsData = (await octokit.search.repos({
+                q: `${terms} org:${orgName} `,
+                per_page: MAX_RETURNED_TARGETS,
+            })).data;
+            if (searchResultsData.total_count === 0) {
+                return [];
+            }
+            searchRepos = searchResultsData.items.map(r => r.full_name);
+        }
+
+        const foundRepos = [];
+        let installationsCount = 0;
+        let totalCount = 0;
+        let page = 1;
+        // Find which ones can be installed by us.
+        do {
+            const { data } = await octokit.apps.listInstallationReposForAuthenticatedUser({
+                installation_id: installationId,
+                page,
+                per_page: 100,
+            });
+            // No results, so stop trying.
+            if (data.repositories.length === 0) {
+                break;
+            }
+            page++;
+            installationsCount += data.repositories.length;
+            totalCount = data.total_count;
+            // Find any repos that were in our search results. If a search term isn't defined, just return it.
+            foundRepos.push(...data.repositories.filter((installRepo) => searchRepos?.includes(installRepo.full_name) ?? true));
+        } while (
+            installationsCount < totalCount &&
+            foundRepos.length < (searchRepos?.length ?? MAX_RETURNED_TARGETS)
```

Comment:
`foundRepos` may contain more items than `MAX_RETURNED_TARGETS` which makes the variable name misleading.

Reply:
I think this is okay. The initial search can return at most `MAX_RETURNED_TARGETS`, and then our filter code can return only items in that search set. There shouldn't be dupes, unless GitHub's API returns dupes.